addon/-private/adapters/rest-adapter.js

```diff
@@ -3,18 +3,18 @@
 */
 
 import Ember from 'ember';
-import Adapter from "ember-data/system/adapter";
+import Adapter from "ember-data/-private/system/adapter";
```

Comment:
Right now in Ember Data 2.2.0 the public api to access any Ember Data module is: 

```js
import DS from 'ember-data';

// then do something with `DS.RESTAdapter`
```

Instead of trying to make a judgement call about what we should allow users to import and what we should keep hidden I decided to move everything into the private directory. After this is merged we can open new prs to move things like the `RESTAdapter` to a public path. That way we can be more deliberate and discerning about what gets exposed publicly.